core/src/main/java/software/amazon/awssdk/util/DateUtils.java

```diff
@@ -152,25 +103,19 @@ private static String tempDateStringForJodaTime(String dateString) {
      * @return The ISO 8601 string representing the specified date.
      */
     public static String formatIso8601Date(Date date) {
-        try {
-            return ISO_8601_DATE_FORMAT.print(date.getTime());
-        } catch (RuntimeException ex) {
-            throw handleException(ex);
-        }
+        return ISO_DATE_TIME.format(ZonedDateTime.ofInstant(date.toInstant(), UTC));
```

Comment:
Better to use `ISO_INSTANT` since we always use UTC anyway. Same for other places where `ISO_DATE_TIME` is use.